jerry-core/ecma/base/ecma-globals.h

```diff
@@ -1216,57 +1215,37 @@ typedef double ecma_number_t;
  */
 #define ECMA_STRING_NOT_ARRAY_INDEX UINT32_MAX
 
-/*
- * Ecma-collection: a growable list of ecma-values. Currently the list is
- * a chain list, where appending new items at the end is cheap operation.
- *
- * Enumerating elements is also cheap, since each page is terminated by a
- * special ecma-value: collection-type. This type has a pointer to the next
- * chunk. The last chunk is terminated by a NULL pointer. There when the
- * next value is requested from the iterator it simply checks the next
- * memory location. If it is not a collection-type value, it returns with
- * the value. Otherwise it gets the start address of the next chunk, and
- * return the value there.
- *
- * The collection-type value is always the last item of a collection chunk,
- * even if the chunk is not completely filled with values (this is only true
- * for the last chunk). Each chunk must have at least one non collection-type
- * value as well.
+/**
+ * Ecma-collection: a growable list of ecma-values.
  */
+typedef struct
+{
+  uint32_t item_count; /**< number of items in the collection */
+  uint32_t capacity; /**< number of items can be stored in the underlying buffer */
+  ecma_value_t *buffer_p; /**< underlying data buffer */
+} ecma_collection_t;
 
 /**
- * Collection flags.
+ * TODO
```

Comment:
It would seem there's still something left to do.

Reply:
Opps, thanks I've added the missing descriptions.